src/V3LinkLevel.cpp

```diff
@@ -59,15 +59,23 @@ void V3LinkLevel::modSortByLevel() {
     if (tops.size() >= 2) {
         const AstNode* const secp = tops[1];  // Complain about second one, as first often intended
         if (!secp->fileline()->warnIsOff(V3ErrorCode::MULTITOP)) {
+            auto warnTopModules = [](std::string warnMore, ModVec tops)
+                                      VL_REQUIRES(V3Error::s().m_mutex) -> std::string {
+                std::stringstream ss;
+                for (AstNode* alsop : tops) {
+                    ss << warnMore << "... Top module " << alsop->prettyNameQ() << endl
+                       << alsop->warnContextSecondary();
+                }
+                return ss.str();
+            };
+
```

Comment:
I don't think warnContextSecondary should need a lock (or at least not the V3Error lock), so you should be able to avoid a lambda and just execute this code here and print ss.str() directly below.

Reply:
This lock is not for ``warnContextSecondary`` but for ``warnMore`` (it is passed as argument to lambda as its value is the same for all modules, but we need to have V3Error lock to make sure we always get correct value)